src/AWS.Deploy.Orchestration/CdkProjectHandler.cs

```diff
@@ -168,27 +196,55 @@ public async Task<bool> DetermineIfCDKBootstrapShouldRun()
             return false;
         }
 
-        private async Task CheckCdkDeploymentFailure(CloudApplication cloudApplication, DateTime deploymentStartDate)
+        private async Task<string> RetreiveStackId(CloudApplication cloudApplication, CancellationToken cancellationToken)
+        {
+            // Do an initial delay to avoid a race condition of the status being checked before the deployment has kicked off.
+            await Task.Delay(TimeSpan.FromSeconds(3));
+
+            await WaitUntilHelper.WaitUntil(async () =>
+            {
+                try
+                {
+                    var stack = await _awsResourceQueryer.GetCloudFormationStack(cloudApplication.Name);
+                    return stack != null;
+                }
+                catch (ResourceQueryException exception) when (exception.InnerException != null && exception.InnerException.Message.Equals($"Stack with id {cloudApplication.Name} does not exist"))
+                {
+                    return false;
+                }
+            }, TimeSpan.FromSeconds(1), TimeSpan.FromMinutes(5), cancellationToken);
```

Comment:
To reduce throttling errors between us and the CDK ping 3 seconds instead of 1.

Reply:
Done